### ISSUE TITLE
Wrong chroma with simple degrain script (regression)

### ISSUE THREAD

**Issue by mkauf** [pinterf/mvtools#26]
This degrain script works with mvtools 2.7.34, but does not work anymore (wrong chroma) starting with mvtools 2.7.35. The current version 2.7.40 also has the bug. I am testing with AviSynth 2.6.0 ST (32-bit build).

```
LoadPlugin("mvtools2.dll")

source = Version()
source = ConvertToYUY2(source)
source = AssumeTFF(AssumeFrameBased(source))

fields = source.SeparateFields() # separate by fields
super = fields.MSuper()
backward_vec2 = super.MAnalyse(isb = true, delta = 2)
forward_vec2 = super.MAnalyse(isb = false, delta = 2)

fields.MDegrain1(super, backward_vec2, forward_vec2)
degrained = Weave()

source = Subtitle(source, "Source")
degrained = Subtitle(degrained, "Degrained")

StackHorizontal(source, degrained)
```

![Bug](https://user-images.githubusercontent.com/12849992/56993484-dd193e00-6b9c-11e9-9fdf-5561fcf8e56c.png)

**Comment by pinterf** [comment]
MSuper is not intended to work with native YUY2. If you keep using YUY2 for some reason instead of YV16, the clip has to be converted to planar with the Interleaved2Planar()/Planar2Interleaved() planar hack.

**Comment by mkauf** [comment]
But MSuper has a parameter called "planar", I think it can be used to specify whether the input is native YUY2 or planar YUY2.

I had a look at the changes between 2.7.34 and 2.7.35. I think there's a small bug in commit e92d88391. On line 212 in MVSuper.cpp:
```
YUY2ToPlanes(src->GetReadPtr(), src->GetPitch(), nWidth, nHeight,
  pSrc[0], nSrcPitch[0], pSrc[1], pSrc[1], nSrcPitch[1], cpuFlags);
```

Fix:
```
YUY2ToPlanes(src->GetReadPtr(), src->GetPitch(), nWidth, nHeight,
  pSrc[0], nSrcPitch[0], pSrc[1], pSrc[2], nSrcPitch[1], cpuFlags);
```

**Comment by pinterf** [comment]
I'm at my PC, and yes, you are right. I was remembering the doc: "Note: super clip is always planar." but that only means that the output of MSuper is left planar for YUY2.

**Comment by mkauf** [comment]
Fixed in release 2.7.41 :-)
